pkg/logs/internal/parsers/integrations/integrations.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package integrations implements a Parser for integrations logs files.
+package integrations
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/internal/parsers"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+)
+
+// New creates a parser that extracts the `ddtags` field from JSON logs, adds
+// them as tags to the log, then submits the rest of the log as is. Non-JSON
+// logs will be submitted as is.
+func New() parsers.Parser {
+	return &integrationFileFormat{}
+}
+
+type integrationFileFormat struct{}
+
+// Parse implements Parser#Parse
+func (p *integrationFileFormat) Parse(msg *message.Message) (*message.Message, error) {
+	// Parse will submit the original message if it encounters an error
+	var data map[string]interface{}
+	err := json.Unmarshal(msg.GetContent(), &data)
+	if err != nil {
+		return msg, err
+	}
+
+	// Step 2: Look for the attribute "ddtags" and read its value
+	ddtagsData, exists := data["ddtags"]
+	if !exists {
+		return msg, nil
+	}
+	ddtagsString, ok := ddtagsData.(string)
+	if !ok {
+		return msg, nil
+	}
+
+	// Step 3: Remove the "ddtags" attribute from the map
+	delete(data, "ddtags")
+
+	// Step 4: Marshal the modified map back to JSON
+	modifiedJSON, err := json.Marshal(data)
+	if err != nil {
+		return msg, err
+	}
+
+	// Step 5: Convert the ddtagsSlice into a string array
+	var ddtagsSlice []string
+	if len(ddtagsString) > 0 {
+		ddtagsSlice = strings.Split(ddtagsString, ",")
+
+		ddtagsSlice = normalizeTags(ddtagsSlice)
+	}
+
+	// append ddtags to message origin tags
+	// set content to modifiedJSON
+	if len(ddtagsSlice) > 0 {
+		msg.ParsingExtra.Tags = ddtagsSlice
```

Comment:
Is this the right place to add the tags to? Should we not append it to `msg.Origin.tags`?
To do so you'll have to do something like:
```
tmpTags := msg.Origin.Tags(ddtagsSlice) // Returns a slice of tags containing Origin tags + ddtagsSlice
msg.Origin.SetTags(tmpTags) // Update Origin with new tags
```

Reply:
So this is actually what I tried at first, but the `msg` type passed to the `Parse` function doesn't have an `origin` field, so when I set it using `msg.Origin.SetTags()` it resulted in a segfault. I suspect the actual object being passed to the parser might be a [`MessageContent`](https://github.com/DataDog/datadog-agent/blob/d3901a226972f5809f380464a2041eb1ab63c69f/pkg/logs/message/message.go#L88).